src/objective/multiclass_obj.cu

```diff
@@ -63,11 +55,14 @@ class SoftmaxMultiClassObj : public ObjFunction {
     const int nclass = param_.num_class;
     const auto ndata = static_cast<int64_t>(preds.Size() / nclass);
 
-    out_gpair->Shard(GPUDistribution::Granular(devices_, nclass));
-    info.labels_.Shard(GPUDistribution::Block(devices_));
-    info.weights_.Shard(GPUDistribution::Block(devices_));
-    preds.Shard(GPUDistribution::Granular(devices_, nclass));
-    label_correct_.Shard(GPUDistribution::Block(devices_));
+    auto devices = GPUSet::All(tparam_->gpu_id, tparam_->n_gpus, preds.Size());
```

Comment:
Why are we constructing new `GPUSet` here? Is this to allow the user to change number of GPUs in the middle of an active Python session?

Reply:
@hcho3 No, user cannot change gpu_id or n_gpus, it's a  known limitation in XGBoost to be solved. But we can switch between GPU and CPU. :)